File: packages/js-dpp/lib/errors/ProtocolVersionParsingError.js
const ConsensusError = require('./ConsensusError');

class ProtocolVersionParsingError extends ConsensusError {
  /**
   * @param {Buffer} payload
   * @param {Error} parsingError
   */
  constructor(payload, parsingError) {
    super(
      `Can't read protocol version from serialized object: ${parsingError.message}`,
    );

    this.payload = payload;
    this.parsingError = parsingError;
  }

  /**
   * Get object payload
   *
   * @return {Buffer}
   */
  getPayload() {
    return this.payload;
  }

  /**
   * Get parsing error
   *
   * @return {Error}
   */
  getParsingError() {
    return this.parsingError;
  }
}

module.exports = ProtocolVersionParsingError;

File: packages/rs-dpp/src/errors/consensus/basic/decode/protocol_version_parsing_error.rs
use crate::consensus::basic::BasicError;
use crate::errors::ProtocolError;
use platform_serialization_derive::{PlatformDeserialize, PlatformSerialize};
use thiserror::Error;

use crate::consensus::ConsensusError;
use bincode::{Decode, Encode};

#[derive(
    Error, Debug, Clone, PartialEq, Eq, Encode, Decode, PlatformSerialize, PlatformDeserialize,
)]
#[error("Can't read protocol version from serialized object: {parsing_error}")]
#[platform_serialize(unversioned)]
pub struct ProtocolVersionParsingError {
    /*

    DO NOT CHANGE ORDER OF FIELDS WITHOUT INTRODUCING OF NEW VERSION

    */
    parsing_error: String,
}

impl ProtocolVersionParsingError {
    pub fn new(parsing_error: String) -> Self {
        Self { parsing_error }
    }

    pub fn parsing_error(&self) -> &str {
        &self.parsing_error
    }
}

impl From<ProtocolVersionParsingError> for ConsensusError {
    fn from(err: ProtocolVersionParsingError) -> Self {
        Self::BasicError(BasicError::ProtocolVersionParsingError(err))
    }
}

impl From<ProtocolVersionParsingError> for u32 {
    fn from(_val: ProtocolVersionParsingError) -> Self {
        0
    }
}
